### PR TITLE
Replace the term "curly bracket" with "curly brace"

### LINKED ISSUE #989: Rename 'bracket' to 'brace' when referring to { and }
StyleCop generally refers to `{` and `}` as "curly brackets". However, Roslyn in general refers to these as "curly braces". In my experience (which is admittedly biased by working in parsing for so long), braces seems to be the more widely used term.

I propose changing all uses (code and documentation) of 'bracket' to 'brace' when referring to curly braces.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #989
Supersedes #1046

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1013UnitTests.cs [hunk 1]
```diff
@@ -134,11 +134,11 @@ public class TestClass
         }
 
         /// <summary>
-        /// Verifies that the analyzer will properly handle closing curly brackets in nested curly brackets.
+        /// Verifies that the analyzer will properly handle closing curly braces in nested curly brackets.
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1013UnitTests.cs]: you missed the 2nd curly brackets usage here

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/LayoutRules/SA1500CodeFixProvider.cs [hunk 2]
```diff
@@ -201,7 +201,7 @@ private static bool IsValidFollowingToken(SyntaxToken nextToken)
 
         private static int DetermineIndentationSteps(IndentationOptions indentationOptions, SyntaxToken token)
         {
-            // For a closing curly bracket use the indentation of the corresponding opening curly bracket
+            // For a closing curly brace use the indentation of the corresponding opening curly bracket
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/LayoutRules/SA1500CodeFixProvider.cs]: You missed the 2nd curly bracket reference here.

**Comment by vweijsters** [comment]
:question: Wouldn't it make sense to use only brace, instead of curly brace, since Roslyn is also using only brace?

**Comment by sharwell** [comment]
@vweijsters Updated!

**Comment by vweijsters** [comment]
:+1: Looks good!

**Pull request merged**